decentraland/unity-renderer
# fix: aggressive thumbnail loading on avatar load

## What does this PR change?
Addresses #1158 

This PR introduces a new way of loading the avatar snapshots through the new `LazyTextureObserver` helper. This helper makes sure a texture is only loaded when a listener marks that the texture is needed. 

Also, if the texture should change because the profile is updated, `LazyTextureObserver` notifies all its listeners of the change.

A few changes on the UI panels that use snapshots were introduced to only load textures when the UI is visible (i.e. enabled). The outcome is that when you open a UI that use them, they will start showing as blank and lazily load over time.

## How to test the changes?
The changes should affect the following HUDs:
* The neighboring users panel that's on the minimap
* Everything related with friends (private conversation, speech bubbles, etc)
* Passport
* Your profile card

The snapshots should unload/load when the HUDs visibility is changed. A good test is to close and open the HUDs and ensure the snapshots load as intended.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
